numpy/lib/histograms.py

```diff
@@ -436,6 +505,10 @@ def _get_bin_edges(a, bins, range, weights):
             bin_type = np.result_type(bin_type, float)
 
         # bin edges must be computed
+        if all([bins is 'auto', auto_method is 'fd', n_equal_bins>2]):
```

Comment:
Test for string equality with `==` : `bins == 'auto', auto_method == 'fd'`

Reply:
Also  use `and` and not `all`, not just easier to read, also faster.